### PR TITLE
feat: fire event when user change the country field

### PR DESCRIPTION
**Pull request by ahtesham-quraish**
Description: Check how many users update country on registration form after we auto populate it with IP. 

[VAN-1793](https://2u-internal.atlassian.net/browse/VAN-1793)

How Has This Been Tested?
It has been tested locally.

### CODE REVIEW

File: src/register/data/reducers.js [hunk 1]
```diff
@@ -108,6 +110,7 @@ const reducer = (state = defaultState, action = {}) => {
         return {
           ...state,
           backendCountryCode: countryCode,
+          IPBasedCountryCode: countryCode,
```
Comment (zainab-amir) [inline src/register/data/reducers.js:113]: why can't we use `backendCountryCode`? Why do we need a new state for it?

File: src/register/RegistrationPage.jsx [hunk 2]
```diff
@@ -177,8 +178,16 @@ const RegistrationPage = (props) => {
...
+      const { country } = configurableFormFields;
+      if (country?.countryCode !== IPBasedCountryCode) {
+        sendTrackEvent('edx.bi.user.auto-populatedcountry.changed', {
+          originalCountry: IPBasedCountryCode,
+          selectedCountry: country.countryCode,
+        });
+      }
     }
-  }, [registrationResult]);
+  }, [registrationResult, IPBasedCountryCode, configurableFormFields]);
```
Comment (zainab-amir) [inline src/register/RegistrationPage.jsx:190]: Just pick country in the dependency array. 
```suggestion
  }, [registrationResult, IPBasedCountryCode, configurableFormFields.country]);
```

File: src/register/RegistrationPage.jsx [hunk 3]
```diff
@@ -177,8 +178,16 @@ const RegistrationPage = (props) => {
 
       // This is used by the "User Retention Rate Event" on GTM
       setCookie(getConfig().USER_RETENTION_COOKIE_NAME, true);
+
```
Comment (zainab-amir) [inline src/register/RegistrationPage.jsx:181]: Add a note stating the temporary nature of this code

File: src/register/data/reducers.js (hunk 1 above)
Comment (ahtesham-quraish) [reply src/register/data/reducers.js:113]: this ```backendCountryCode``` is reset when ```BACKUP_REGISTRATION_DATA.BEGIN``` is fired so i did not touch that state I created new one which does not get reset when above action is fired.

File: src/register/RegistrationPage.jsx (hunk 2 above)
Comment (ahtesham-quraish) [reply src/register/RegistrationPage.jsx:190]: addressed

File: src/register/RegistrationPage.jsx (hunk 3 above)
Comment (ahtesham-quraish) [reply src/register/RegistrationPage.jsx:181]: I have added the note please see if it makes sense

File: src/register/RegistrationPage.jsx (hunk 2 above)
Comment (ahtesham-quraish) [reply src/register/RegistrationPage.jsx:193]: I have tried to use it but there is lint error which says ```configurableFormFields``` is not in a dependency array

Comment (zainab-amir) [reply src/register/RegistrationPage.jsx:193]: Remove line number 185 and directly us the const `configurableFormFields.country` value

Comment (ahtesham-quraish) [reply src/register/RegistrationPage.jsx]: yeah that makes sense

**Review by zainab-amir — APPROVED** [review]

**Pull request merged**